app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt

```diff
@@ -34,29 +39,28 @@ class SecondFragment : BaseFragment() {
         viewModel.output.persistData bindTo ::bindPersistedData
     }
 
-//    @NeedsPermission(Manifest.permission.CAMERA)
-//    fun openCamera() {
-//        startActivity(Intent(MediaStore.ACTION_IMAGE_CAPTURE))
-//    }
-//
-//    @OnPermissionDenied(Manifest.permission.CAMERA)
-//    fun showDeniedForCamera() {
-//        Toast.makeText(this, "Permission camera denied", Toast.LENGTH_SHORT).show()
-//    }
-//
-//    @OnNeverAskAgain(Manifest.permission.CAMERA)
-//    fun showNeverAskForCamera() {
-//        Toast.makeText(this, "Permission camera never ask", Toast.LENGTH_SHORT).show()
-//    }
+    private fun requestCamera() {
+        rxPermission
+            .requestEach(Manifest.permission.CAMERA)
+            .subscribe(::handleCameraPermission)
+            .addToDisposables()
+    }
 
-//    override fun onRequestPermissionsResult(
-//        requestCode: Int,
-//        permissions: Array<out String>,
-//        grantResults: IntArray
-//    ) {
-//        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-//        onRequestPermissionsResult(requestCode, permissions, grantResults)
-//    }
+    private fun handleCameraPermission(permission: Permission) {
+        when {
+            permission.granted -> {
+                startActivity(Intent(MediaStore.ACTION_IMAGE_CAPTURE))
+            }
+            permission.shouldShowRequestPermissionRationale -> {
+                // Deny
+                toaster.display("Permission camera denied")
+            }
+            else -> {
+                // Deny and never ask again
+                toaster.display("Permission camera never ask")
```

Comment:
No biggie: "Permission camera never ask again"

Reply:
or `Camera permission is never asked again` 😁